SharpPcap/LibPcap/PcapInterface.cs

```diff
@@ -131,6 +115,34 @@ internal PcapInterface(pcap_if pcapIf, NetworkInterface networkInterface, ICrede
 
                 address = addr.Next; // move to the next address
             }
+
+            // attempt to populate the mac address,
+            // friendly name etc of this device
+            if (networkInterface != null)
+            {
+                var ipProperties = networkInterface.GetIPProperties();
+                int gatewayAddressCount = ipProperties.GatewayAddresses.Count;
+                if (gatewayAddressCount != 0)
+                {
+                    foreach (GatewayIPAddressInformation gatewayInfo in ipProperties.GatewayAddresses)
+                    {
+                        GatewayAddresses.Add(gatewayInfo.Address);
+                    }
+                }
+                FriendlyName = networkInterface.Name;
+
+                PhysicalAddress mac = networkInterface.GetPhysicalAddress();
+                if (MacAddress == null && mac != null)
```

Comment:
`GetPhysicalAddress()` never returns null, it may return a `PhysicalAddress` with 0 bytes, or an address with 6 zeros. but never null.
See:
* https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs#L14
* https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/UnixNetworkInterface.cs#L14
* https://github.com/dotnet/runtime/blob/98eee0587dfd0752c030fe49b3fd9b45f4b2e5c5/src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemNetworkInterface.cs#L177
* https://referencesource.microsoft.com/#System/net/System/Net/NetworkInformation/SystemNetworkInterface.cs,165

Reply:
@kayoub5 the public API doesn't make any commitments to never returning null, https://docs.microsoft.com/en-us/dotnet/api/system.net.networkinformation.networkinterface.getphysicaladdress?view=netcore-3.1

I'd be inclined to protect against the chance (allowed by the public api docs) that they can return null for now. If they clarify the public api we can always drop the check.

Reply:
Sounds good.